### PR TITLE
Style and workflow fixes for the main screen.

### PR DESCRIPTION
**Pull request by jgraham**
[Review on Reviewable](https://reviewable.io/reviews/jgraham/wptview/38)

### CODE REVIEW

File: LovefieldService.js [hunk 1]
```diff
@@ -402,12 +401,21 @@ LovefieldService.prototype.selectFilteredResults = function(filters, pathFilters
...
-  return tx.exec([q1, q2, q3]);
+LovefieldService.prototype.deleteEntries = function(run_id) {
+  return this.getDbConnection().then((db) => {
+    var q1 = db.delete().from(this.test_results);
+    var q2 = db.delete().from(this.tests);
+    var q3 = db.delete().from(this.test_runs);
+    if (run_id) {
+      q1 = q1.where(this.test_results.run_id.eq(run_id));
+      q2 = q2
+        .leftOuterJoin(this.test_results, this.test_results.test_id.eq(this.tests.test_id))
```
Comment (martiansideofthemoon) [inline LovefieldService.js]: I think this should be `this.tests.id`. I can't clear the results

File: LovefieldService.js [hunk 2]
```diff
@@ -402,12 +401,21 @@ LovefieldService.prototype.selectFilteredResults = function(filters, pathFilters
...
 
-LovefieldService.prototype.deleteEntries = function() {
-  var q1 = this.db_.delete().from(this.test_results);
-  var q2 = this.db_.delete().from(this.tests);
-  var q3 = this.db_.delete().from(this.test_runs);
-  var tx = this.db_.createTransaction();
-  return tx.exec([q1, q2, q3]);
+LovefieldService.prototype.deleteEntries = function(run_id) {
+  return this.getDbConnection().then((db) => {
+    var q1 = db.delete().from(this.test_results);
```
Comment (martiansideofthemoon) [inline LovefieldService.js]: Maybe we could create variables named `tests`, `test_results` and `test_runs` for consistency with the other functions.
Something like `var tests = this.tests` and so on

File: wptview.css [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    padding: 0.5em;
+    background-color: #e7ebf1;
+}
+
+div#wptview {
+    padding-left: 16px;
+    padding-right: 16px;
+}
+
+table {
```
Comment (martiansideofthemoon) [inline wptview.css]: Maybe we could move the table to the right of the import sequence. I find it inconvenient to scroll so much. What do you feel?

File: wptview.css [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    padding: 0;
+    margin:0;
+    background-color: #f8f8f8;
+}
+
+body {
+    margin: 0;
+}
+
+h1 {
```
Comment (martiansideofthemoon) [inline wptview.css]: The interface looks a lot better than before, except this. Maybe a font change will help here?

**Comment by martiansideofthemoon** [comment]
Review status: 0 of 4 files reviewed at latest revision, 12 unresolved discussions.

---

_[angular_scripts.js, line 84 [r1]](https://reviewable.io:443/reviews/jgraham/wptview/38#-K5_z1FezFp7ZXZjkian:-K5_z1FezFp7ZXZjkiao:1527893630) ([raw file](https://github.com/jgraham/wptview/blob/b0b692982558bfea61363dd26351beb3053bf3b1/angular_scripts.js#L84)):_
`;` for consistency?

---

_[index.html, line 6 [r1]](https://reviewable.io:443/reviews/jgraham/wptview/38#-K5_tilVzghsjgKhvUf7:-K5_tilVzghsjgKhvUf8:935633722) ([raw file](https://github.com/jgraham/wptview/blob/b0b692982558bfea61363dd26351beb3053bf3b1/index.html#L6)):_
**nit** : Maybe putting this as WPT will help the aesthetics?

---

_[index.html, line 20 [r1]](https://reviewable.io:443/reviews/jgraham/wptview/38#-K5_v6mUfK-Lkmflu4P5:-K5_v6mUfK-Lkmflu4P6:158795509) ([raw file](https://github.com/jgraham/wptview/blob/b0b692982558bfea61363dd26351beb3053bf3b1/index.html#L20)):_
Is this needed? For a zero length `runs`, won't this be invisible?

---

_[index.html, line 31 [r1]](https://reviewable.io:443/reviews/jgraham/wptview/38#-K5_vWVRgXZt7Ad8WXPz:-K5_vWVRgXZt7Ad8WXQ-:1845893413) ([raw file](https://github.com/jgraham/wptview/blob/b0b692982558bfea61363dd26351beb3053bf3b1/index.html#L31)):_
Could we make this a button? I find the "Import" text getting selected each time we run this

---

_[index.html, line 46 [r1]](https://reviewable.io:443/reviews/jgraham/wptview/38#-K5_wx8jdzzoIDmJFC9L:-K5_wx8jdzzoIDmJFC9M:-1568368757) ([raw file](https://github.com/jgraham/wptview/blob/b0b692982558bfea61363dd26351beb3053bf3b1/index.html#L46)):_
As a user I would probably prefer having this button only clear the table and not the database.

---

_[index.html, line 60 [r1]](https://reviewable.io:443/reviews/jgraham/wptview/38#-K5_xj67cobdsqPV_yWN:-K5_xj68DGaMWUm-ECft:1984349252) ([raw file](https://github.com/jgraham/wptview/blob/b0b692982558bfea61363dd26351beb3053bf3b1/index.html#L60)):_
`ng-disabled="busy"` here as well?

---

_[LovefieldService.js, line 413 [r1]](https://reviewable.io:443/reviews/jgraham/wptview/38#-K5a0z0QwEpd182dvzgi:-K5a0z0QwEpd182dvzgj:106400814) ([raw file](https://github.com/jgraham/wptview/blob/b0b692982558bfea61363dd26351beb3053bf3b1/LovefieldService.js#L413)):_
I'm not very sure about this one. Are you sure that this will delete all those tests that are exclusively for `run_id`? Or am I misinterpreting? 

---

_[LovefieldService.js, line 450 [r1]](https://reviewable.io:443/reviews/jgraham/wptview/38#-K5a2P9Ec-o1JdKUBbz0:-K5a2P9Ec-o1JdKUBbz1:-1752153630) ([raw file](https://github.com/jgraham/wptview/blob/b0b692982558bfea61363dd26351beb3053bf3b1/LovefieldService.js#L450)):_
**nit**: Maybe we could keep a column alias for `COUNT('result_id')` for clarity

---

_[wptview.css, line 5 [r1]](https://reviewable.io:443/reviews/jgraham/wptview/38#-K5_z_FDG0K1vaIHZPc4:-K5_z_FDG0K1vaIHZPc5:-644461087) ([raw file](https://github.com/jgraham/wptview/blob/b0b692982558bfea61363dd26351beb3053bf3b1/wptview.css#L5)):_
**nit**: Space between `:` and 0

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/jgraham/wptview/38)_

**Comment by martiansideofthemoon** [comment]
Review status: 0 of 4 files reviewed at latest revision, 13 unresolved discussions.

---

_[LovefieldService.js, line 406 [r1]](https://reviewable.io:443/reviews/jgraham/wptview/38#-K5a4IFRochtrsmwfG96:-K5a4IFRochtrsmwfG97:-882210944) ([raw file](https://github.com/jgraham/wptview/blob/b0b692982558bfea61363dd26351beb3053bf3b1/LovefieldService.js#L406)):_
Maybe we could create variables named `tests`, `test_results` and `test_runs` for consistency with the other functions.
Something like `var tests = this.tests` and so on

---

_[LovefieldService.js, line 412 [r1]](https://reviewable.io:443/reviews/jgraham/wptview/38#-K5a4QmDL0e10tyIpNfe:-K5a4QmDL0e10tyIpNff:-1399099227) ([raw file](https://github.com/jgraham/wptview/blob/b0b692982558bfea61363dd26351beb3053bf3b1/LovefieldService.js#L412)):_
`this.tests.id`. I can't clear the results

---

_[wptview.css, line 13 [r1]](https://reviewable.io:443/reviews/jgraham/wptview/38#-K5a46DD2XzuR_vj4lt9:-K5a46DEe9X5wBq1Wfbn:356249192) ([raw file](https://github.com/jgraham/wptview/blob/b0b692982558bfea61363dd26351beb3053bf3b1/wptview.css#L13)):_
The interface looks a lot better than before, except this. Maybe a font change will help here?

---

_[wptview.css, line 26 [r1]](https://reviewable.io:443/reviews/jgraham/wptview/38#-K5a4ACTDPdw_rN4LLpT:-K5a4ACTDPdw_rN4LLpU:-1693161169) ([raw file](https://github.com/jgraham/wptview/blob/b0b692982558bfea61363dd26351beb3053bf3b1/wptview.css#L26)):_
Maybe we could move the table to the right of the import sequence. I find it inconvenient to scroll so much. What do you feel?

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/jgraham/wptview/38)_

**Comment by martiansideofthemoon** [comment]
Reviewed 4 of 4 files at r1.
Review status: all files reviewed at latest revision, 13 unresolved discussions.

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/jgraham/wptview/38)_

**Comment by jgraham** [comment]
Review status: 0 of 4 files reviewed at latest revision, 6 unresolved discussions.

---

_[index.html, line 31 [r1]](https://reviewable.io:443/reviews/jgraham/wptview/38#-K5_vWVRgXZt7Ad8WXPz:-K5aN7F0wKIPxM5geQxg:-772536722) ([raw file](https://github.com/jgraham/wptview/blob/b0b692982558bfea61363dd26351beb3053bf3b1/index.html#L31)):_
I don't think a button works well because this is stateful. If we had a control that would stay in an active state after clicking once we could perhaps use that, but I'm not sure we do. Have a look again with the next push which puts the controls on the side.

---

_[index.html, line 46 [r1]](https://reviewable.io:443/reviews/jgraham/wptview/38#-K5_wx8jdzzoIDmJFC9L:-K5aNXcKohFF-ETELVuS:1213056003) ([raw file](https://github.com/jgraham/wptview/blob/b0b692982558bfea61363dd26351beb3053bf3b1/index.html#L46)):_
I don't understand? It seems pretty useful to be able to discard all the data that you previously loaded if you want to move onto a new comparison with some different logs.

---

_[LovefieldService.js, line 413 [r1]](https://reviewable.io:443/reviews/jgraham/wptview/38#-K5a0z0QwEpd182dvzgi:-K5aOSnjRyYRvIikOGj8:-1687043391) ([raw file](https://github.com/jgraham/wptview/blob/b0b692982558bfea61363dd26351beb3053bf3b1/LovefieldService.js#L413)):_
Yeah, but it doesn't work. I made a fix.

---

_[wptview.css, line 26 [r1]](https://reviewable.io:443/reviews/jgraham/wptview/38#-K5a4ACTDPdw_rN4LLpT:-K5aOj7WH8ey3k6j4etd:948092168) ([raw file](https://github.com/jgraham/wptview/blob/b0b692982558bfea61363dd26351beb3053bf3b1/wptview.css#L26)):_
Seems reasonable to rearrange things a bit. I'll look.

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/jgraham/wptview/38)_

**Comment by martiansideofthemoon** [comment]
Review status: 0 of 4 files reviewed at latest revision, 3 unresolved discussions.

---

_[LovefieldService.js, line 423 [r2]](https://reviewable.io:443/reviews/jgraham/wptview/38#-K5ao4zXUdycZ0APKHlj:-K5ao4zXUdycZ0APKHlk:761624784) ([raw file](https://github.com/jgraham/wptview/blob/5a665e576cce28285a833278450d7c55c8a496dc/LovefieldService.js#L423)):_
**nit**: I suppose the queries are trying to delete the `tests` which don't have any `results`. As it's the last part of a transaction, it will delete all the `tests` which no longer have results once the `results` corresponding to `run_id` have been deleted. For code clarity, could we move this out of the transaction and into the `then()` for `tx`? It's not a necessity, and if it's affecting a lot of code (for the case where `run_id` in not present) you may ignore this. It's just that I took some time to comprehend why it works. 

---

_[LovefieldService.js, line 431 [r2]](https://reviewable.io:443/reviews/jgraham/wptview/38#-K5amgnzJEIbxrPgMG-H:-K5amgo-2fLOEp4hdPff:-1682437961) ([raw file](https://github.com/jgraham/wptview/blob/5a665e576cce28285a833278450d7c55c8a496dc/LovefieldService.js#L431)):_
**nit**: We won't need the `{ }` here right?

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/jgraham/wptview/38)_

**Comment by martiansideofthemoon** [comment]
Reviewed 4 of 4 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/jgraham/wptview/38)_

**Comment by jgraham** [comment]
Fixes #31, #17, #15

**Comment by martiansideofthemoon** [comment]
Review status: 3 of 4 files reviewed at latest revision, 5 unresolved discussions.

---

_[angular_scripts.js, line 124 [r4]](https://reviewable.io:443/reviews/jgraham/wptview/38#-K5gcNcoTANQ5yY1GBXz:-K5gcNcoTANQ5yY1GBY-:-1206152399) ([raw file](https://github.com/jgraham/wptview/blob/7e614e1ed52732bb4935bba3ccdbd41cc629dea3/angular_scripts.js#L124)):_
After I clear the results, I still see some warnings. We need to update add a `$scope.warning_message = ""` or something here.

---

_[LovefieldService.js, line 420 [r4]](https://reviewable.io:443/reviews/jgraham/wptview/38#-K5gZHf3xfJYgOiRnZHw:-K5gZHf3xfJYgOiRnZHx:-1051067598) ([raw file](https://github.com/jgraham/wptview/blob/7e614e1ed52732bb4935bba3ccdbd41cc629dea3/LovefieldService.js#L420)):_
**nit**: Could we change this variable name? `tx_promise` maybe?

---

_[LovefieldService.js, line 431 [r4]](https://reviewable.io:443/reviews/jgraham/wptview/38#-K5gXt3jLHd1fGz1tGat:-K5gXt3kRCSMmQlfGrrz:-490234959) ([raw file](https://github.com/jgraham/wptview/blob/7e614e1ed52732bb4935bba3ccdbd41cc629dea3/LovefieldService.js#L431)):_
**nit**: `query` spelled incorrectly, and below too

---

_[LovefieldService.js, line 439 [r4]](https://reviewable.io:443/reviews/jgraham/wptview/38#-K5gYkUe54-Uph4Mgawf:-K5gYkUe54-Uph4Mgawg:-1896122716) ([raw file](https://github.com/jgraham/wptview/blob/7e614e1ed52732bb4935bba3ccdbd41cc629dea3/LovefieldService.js#L439)):_
**nit**: Make it a `forEach` for consistency?

---

_[LovefieldService.js, line 452 [r4]](https://reviewable.io:443/reviews/jgraham/wptview/38#-K5gZD2sk0o11F4JKzUn:-K5gZD2sk0o11F4JKzUo:1771528061) ([raw file](https://github.com/jgraham/wptview/blob/7e614e1ed52732bb4935bba3ccdbd41cc629dea3/LovefieldService.js#L452)):_
**nit**: `;` for consistency and 3 lines after.

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/jgraham/wptview/38)_

**Comment by martiansideofthemoon** [comment]
Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/jgraham/wptview/38)_

**Comment by martiansideofthemoon** [comment]
Reviewed 3 of 3 files at r5.
Review status: all files reviewed at latest revision, all discussions resolved.

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/jgraham/wptview/38)_

**Comment by martiansideofthemoon** [comment]
**r+**, squash and then we can merge!

**Pull request merged**